### PR TITLE
update Hugo

### PR DESCRIPTION
**Pull request by maelle**
Updating Hugo is a good idea (there are new features like the one I want to use in #653; contributors might install Hugo latest version locally). However since last time we updated it

* The Markdown renderer changed from Blackfriday to Goldmark. I didn't find any problem with that.

* The syntax highlighting happen on the server side by default which sounds great IMO but I had to transfer the CSS from hljs stuff to Chroma stuff.

Here's what I did to fix the CSS since we have our own style as opposed to one of Chroma style (Chroma styles are like Pygments styles) cf https://gohugo.io/content-management/syntax-highlighting/#generate-syntax-highlighter-css

* Remove the auto-syntax partial and its use, and the parts of code.pcss related to hljs/any sort of coloring.

* Add `pygmentsUseClasses=true` to the site config.

* Generate the CSS for the monokai style so I could amend it.

I wrote a script to find the unique monokai colors.

```r
library("magrittr")
readLines("themes/ropensci/static/css/code.css") %>%
  glue::glue_collapse(sep = " ") %>%
  strsplit(split = " ") %>%
  unlist() %>%
  stringr::str_remove_all("\\}") %>%
  stringr::str_remove_all("\\;") %>%
  purrr::keep(stringr::str_detect, "\\#") %>%
  unique() %>%
  clipr::write_clip()
```

I then looked them up in the code.css and had a look at our old code.pcss to find equivalents. There's no one-to-one mapping of the terms used, so it was a bit trial-and-error. I now realize I could look for an actual comparison table. (well I have now tried, I didn't find one via a search engine :upside_down_face: )

Then I `npm run css` and rendered the site. I fixed a few colors.

I compared some current pages with some new pages to see whether the code looked ok.

### CODE REVIEW

**Comment by maelle** [comment]
The code chunks look weird cf e.g. https://deploy-preview-651--ropensci.netlify.com/technotes/2020/03/19/parzer/

**Comment by maelle** [comment]
Ok I see what needs to be done https://gohugo.io/content-management/syntax-highlighting/ (our own style with the correct CSS. not possible to override via current css because the styling by hljs ends up being an inline styling?!)

**Comment by maelle** [comment]
https://deploy-preview-651--ropensci.netlify.com/technotes/2019/12/08/precompute-vignettes/ vs https://ropensci.org/technotes/2019/12/08/precompute-vignettes/ The margin looks smaller but since a bigger redesign is coming I don't think fiddling with that is worth it.

https://deploy-preview-651--ropensci.netlify.com/technotes/2020/03/19/parzer/ vs https://ropensci.org/technotes/2020/03/19/parzer/ "library" is now purple, is that ok? warnings look different, but it's because the warning block was created with indentation, not code fences, and that's related to how the Rmd was knit (not via our recommended method I'm afraid), so I don't think it'll come up often.

https://deploy-preview-651--ropensci.netlify.com/technotes/2020/02/25/opentripplanner/ vs https://ropensci.org/technotes/2020/02/25/opentripplanner/ no further issue.

**Comment by maelle** [comment]
FTR I love the monokai style but it would obviously not pair well with our color scheme. :grin:

File: themes/ropensci/layouts/404.html [hunk 1]
```diff
@@ -21,9 +21,7 @@ <h2 style="text-align: center;">Missing page</h2>
         <div class="row">
             <div class="col-8 col-offset-2 top-4 bottom-5">
 
-                 <pre><code class="language-r"> <big> page$browse()
-#&gt; Error: object 'page' not found
-</big></code></pre>
+{{ readFile "404.md" | markdownify  }}
```
Comment (maelle) [inline themes/ropensci/layouts/404.html:24]: otherwise the 404 page code would have no highlighting.

**Comment by maelle** [comment]
We _could_ disable server-side syntax highlighting, if there are really good reasons to do that.

**Comment by maelle** [comment]
@sckott old html use case that loses highlighting (because of its not being Markdown content. Chroma highlighting is for code fences in Markdown content) https://deploy-preview-651--ropensci.netlify.com/usecases/rgbif_nichemodel/

**Comment by maelle** [comment]
I'm adding a post example to this PR, that I'll remove before merging, to show how to use the highlighting options (number lines, highlight lines) in chunks cf https://deploy-preview-651--ropensci.netlify.com/blog/2020/04/05/syntax-highlighting/

**Comment by stefaniebutland** [comment]
I removed blog-editors as reviewer so it's not on Steffi's list. I don't think I need to chime in.

**Comment by maelle** [comment]
I'll actually do that fix myself @sckott this way I can see whether I need to add some CSS for "pre". Thanks for the review!

**Pull request merged**